### PR TITLE
Release pipe line ref #119

### LINKED ISSUE #119: Automate hw-go release process in Travis
**Feature description**

Automate release process steps in Travis script

**Describe the solution you'd like**
Add `deploy` sections in `.travis.yml` script

**Describe alternatives you've considered**
Manual release is not an option

**Additional context**
see skycoin/hardware-wallet#183 as a reference

**Possible implementation**

- [x] Add `make release` target
- [x] Add deploy step for Travis jobs
- [x] Invoke `make release` when Travis build has been triggered by a git tag.
- [x] Release matrix build for `osx` and `linux`
- [x] Ensure README is in sync with release script
- [x] Deploy to Github Releases

### PR DESCRIPTION
**Pull request by stdevAlDen**
Fixes #119

Changes:
- Add travis CD instructions for github releases.

Does this change need to mentioned in CHANGELOG.md?
yes

Requires changes in protobuf specs?
no

Requires testing
no

https://github.com/skycoin/hardware-wallet-go/pull/121#issuecomment-490277348

### CODE REVIEW

**Comment by therealssj** [comment]
@stdevAlDen the test is failing because of the lock I introduced in #115 and is not related to ur changed. 
I have a temporary fix in #116 but if u want to merge this PR feel free to just remove the lock.
I will push a proper fix in #116.
sorry for the trouble.

**Comment by stdevAlDen** [comment]
> @stdevAlDen the test is failing because of the lock I introduced in #115 and is not related to ur changes.
> I have a temporary fix in #116 but if u want to merge this PR feel free to just remove the lock.
> I will push a proper fix in #116.
> sorry for the trouble.

@therealssj I no think so, I was able to find [firmware as source of the fail](https://github.com/skycoin/hardware-wallet/issues/215), what do you think?

**Comment by stdevAlDen** [comment]
> > @stdevAlDen the test is failing because of the lock I introduced in #115 and is not related to ur changes.
> > I have a temporary fix in #116 but if u want to merge this PR feel free to just remove the lock.
> > I will push a proper fix in #116.
> > sorry for the trouble.
> 
> @therealssj I no think so, I was able to find [firmware as source of the fail](https://github.com/skycoin/hardware-wallet/issues/215), what do you think?

@iketheadore I means, maybe there is a bug in th #115 but are you agrees to the point that there is a bug in the firmware?

**Comment by therealssj** [comment]
> > @stdevAlDen the test is failing because of the lock I introduced in #115 and is not related to ur changes.
> > I have a temporary fix in #116 but if u want to merge this PR feel free to just remove the lock.
> > I will push a proper fix in #116.
> > sorry for the trouble.
> 
> @therealssj I no think so, I was able to find [firmware as source of the fail](https://github.com/skycoin/hardware-wallet/issues/215), what do you think?

the code indeed looks incorrect.

**Comment by stdevAlDen** [comment]
@therealssj ok, thank you

**Comment by stdevAlDen** [comment]
@olemis & @therealssj thi depennds on:
 - https://github.com/skycoin/hardware-wallet-go/pull/123
 - https://github.com/skycoin/hardware-wallet/pull/221

**Comment by olemis** [comment]
[Travis failure](https://travis-ci.com/skycoin/hardware-wallet-go/builds/110780418#L3683)

**Comment by therealssj** [comment]
@stdevAlDen conflicts

**Comment by stdevAlDen** [comment]
> @stdevAlDen conflicts

@therealssj thank you, solved

File: .travis.yml [hunk 1]
```diff
@@ -73,3 +73,18 @@ after_script:
...
+- provider: releases
+  api_key:
+    secure: ee6xAJoQIXKy8w7DvSMWTpIVymJCzV4raj7MatahyEAtIuQZ0oGR35mxxTHC6/iEkVvOP8+i+g7LTnGgTc1OSJmSq9PwwG+3aNnBLtZbdEs/b+tKkA4xLoK3aRpWJ9dD2uusjI3z3gVqj3N6u3Y7idYPdMSC7+L0G72CdtL9bqjL2j2rStgv5Teb8i9P6uAPDl01uoN1mMBxfGoYES1i+MkRVKg+6In5V+eoar0h1o3zki33W5S0bGSJo3VbasCLoJIiTEDgcnM2rqfJeb07URbSLOBqhFOidVFi9QdfRVfMoRWYmCtiYKyXraCj6LgWbD+A90ZMfOJBBI6iTl2QRsNVge1Jz02MTofH2VGmOpvM7oKmkmvhHdNXnEH7zr1xSNpWDHobvY0Pl6aVoiwK3C+idk8SSfeohEJEuj5uTLRrf3qrafdcMx/YKCYHDyvhxG+zTy5I4+hwxIfT40juUqQkcUEZvsaf0aOKX25EO5aRTnQpkn9rWMytohmBSnm8D5SMlUgrhn3CVO8xysW+VlomAVPEDAVyqopiOJrsD5E8lnp0VuwA4NUC/Gp+ZrxYYgJhAX+utWtVYQ7I4egZ8lIzE3BMJRUxK4ZunlqHsiVqI7nrGEZHvxc2+ALp8mpsfWf6H4t1kfkHWt354cupSl8Z2UlwPW0fE2/XNjE6WSk=
+  file:
+  - skycoin-hw-cli-$(uname -s)-v$(VERSION_RAW)
+  skip_cleanup: true
+  draft: true
+  on:
+    repo: skycoin/hardware-wallet-go
+    tags: true
```
Comment (therealssj) [inline .travis.yml]: newline

File: Makefile [hunk 2]
```diff
@@ -5,6 +5,12 @@
 .PHONY: clean lint check format
 
 GOPATH  ?= $(HOME)/go
+UNAME_S ?= $(shell uname -s)
+VERSION_RAW  =$(shell ./ci-scripts/version.sh)
```
Comment (olemis) [inline Makefile]: pls revert

File: .travis.yml [hunk 3]
```diff
@@ -73,3 +74,19 @@ after_script:
...
+
+before_deploy:
+- make release
+
+deploy:
+- provider: releases
+  api_key:
+    secure: ee6xAJoQIXKy8w7DvSMWTpIVymJCzV4raj7MatahyEAtIuQZ0oGR35mxxTHC6/iEkVvOP8+i+g7LTnGgTc1OSJmSq9PwwG+3aNnBLtZbdEs/b+tKkA4xLoK3aRpWJ9dD2uusjI3z3gVqj3N6u3Y7idYPdMSC7+L0G72CdtL9bqjL2j2rStgv5Teb8i9P6uAPDl01uoN1mMBxfGoYES1i+MkRVKg+6In5V+eoar0h1o3zki33W5S0bGSJo3VbasCLoJIiTEDgcnM2rqfJeb07URbSLOBqhFOidVFi9QdfRVfMoRWYmCtiYKyXraCj6LgWbD+A90ZMfOJBBI6iTl2QRsNVge1Jz02MTofH2VGmOpvM7oKmkmvhHdNXnEH7zr1xSNpWDHobvY0Pl6aVoiwK3C+idk8SSfeohEJEuj5uTLRrf3qrafdcMx/YKCYHDyvhxG+zTy5I4+hwxIfT40juUqQkcUEZvsaf0aOKX25EO5aRTnQpkn9rWMytohmBSnm8D5SMlUgrhn3CVO8xysW+VlomAVPEDAVyqopiOJrsD5E8lnp0VuwA4NUC/Gp+ZrxYYgJhAX+utWtVYQ7I4egZ8lIzE3BMJRUxK4ZunlqHsiVqI7nrGEZHvxc2+ALp8mpsfWf6H4t1kfkHWt354cupSl8Z2UlwPW0fE2/XNjE6WSk=
+  file:
+  - skycoin-hw-cli-$(uname -s)-v$(VERSION_RAW)
```
Comment (olemis) [inline .travis.yml]: where's the definition of `VERSION_RAW` ?

File: Makefile (hunk 2 above)
Comment (olemis) [reply Makefile]: not necessary after 0f18325

**Pull request merged**